### ISSUE TITLE
Rewriting top cell <T> results in ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by andreiarusoaie** [kframework/k-legacy#2212]
Consider the following definition:

```
module EXP-SYNTAX

  // grammar: BNF = Backus-Naur Form
  syntax Exp ::= Int
               | Exp "+" Exp [left, strict]
           | "(" Exp ")" [bracket]

endmodule

module EXP
  imports EXP-SYNTAX

  syntax KResult ::= Int

  configuration <T>
                  <k> $PGM </k>
        </T>

  rul <T> <k> I1:Int + I2:Int </k> </T>
       => 
       <T> <k> I1 +Int I2 </k> </T>

endmodule
```

Then, `kompile`:

```
:-$ kompile exp.k 
[Error] Internal: Uncaught exception thrown of type
ArrayIndexOutOfBoundsException.
Please rerun your program with the --debug flag to generate a stack trace, and
file a bug report at https://github.com/kframework/k/issues
```

Rewriting only the `<k>` cell works. Probably rewriting the top cell is forbidden, but we need a better error message.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/148313234996210/192281468133415)

**Comment by grosu** [comment]
It should work to rewrite the top cell, too, so something is definitely wrong here.

**Comment by cos** [comment]
fixed via #2232
